pkg/manager/impl/execution_manager.go

```diff
@@ -743,6 +738,15 @@ func (m *ExecutionManager) TerminateExecution(
 		logger.Infof(ctx, "couldn't find execution [%+v] to save termination cause", request.Id)
 		return nil, err
 	}
+
+	err = m.workflowExecutor.TerminateWorkflowExecution(ctx, workflowengineInterfaces.TerminateWorkflowInput{
```

Comment:
is this intentional? by design we've intentionally saved in the db only after we successfully deleted the workflow in kubernetes

Reply:
Yes. It's the same.

I am only doing a Get before termination, as I need the cluster where the execution was kicked off